### PR TITLE
Add VictorOps notifier

### PR DESCRIPTION
**Pull request by mboorstin**

### CODE REVIEW

File: notifier/victorops-notifier.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+type VictorOpsEvent struct {
+	// Explicitly listed by http://victorops.force.com/knowledgebase/articles/Integration/Alert-Ingestion-API-Documentation/
+	MessageType       string `json:"message_type"`
+	EntityId          string `json:"entity_id"`
+	Timestamp         uint32 `json:"timestamp"`
+	StateMessage      string `json:"state_message"`
+	MonitoringTool    string `json:"monitoring_tool"`
+	EntityDisplayName string `json:"entity_display_name"`
```
Comment (donaldguy) [inline notifier/victorops-notifier.go:25]: Pretty sure these are all the default serializer tag translations?

but having them explicit doesn't hurt.

**Comment by donaldguy** [comment]
I don't see anything especially objectionable.

There is some flexibility that one might want in terms of not having to use the same routing key for every check - but then thats more a deficiency of consul-alerts as a whole

Additionally, there seems to be an assumption that only service checks are present, but consul lets you define health checks on nodes as a whole, not tied to a specific service. Again this might be an inflexibility of consul-alerts as a whole; but maybe its on you? and maybe its nothing and the use of `Service` is a misnomer just meaning check; I'd need to dig into the consul-alerts source more as a whole

My under the weather brain isn't the most detail oriented this afternoon

**Comment by donaldguy** [comment]
(One can in VO define different rotations for different people by routing key - we don't really do this yet, but may in the future, and other customers of theirs certainly do)

**Comment by donaldguy** [comment]
Hey @jasonhand, do you or someone you can tap at VO have any thing to add before @mboorstin makes an upstream PR?

**Comment by donaldguy** [comment]
It indeed appears to be the case that the Message struct passes both Service and Check info through; If the Check is not of a service is that gonna just be `""`? Would probably require an (easy) test or a look at the watch code in consul itself

I'm pretty sure the opsgenie notifier in consul-alerts also neglects this case.

**Comment by mboorstin** [comment]
Service / Check: Yeah, you can see, for instance, https://github.com/tulip/consul-alerts/blob/master/notifier/pagerduty-notifier.go#L30 explicitly checks that the service exists.  I'll do the same and only pass along `consul_service` and `consul_service_id` to VO if so.
Routing key: Yeah, makes sense that it would be nice for it to be flexible, but as you said as of now there isn't a whole lot of support for it in the rest of consul-alerts.  I might do a separate PR to add in a whole infrastructure for that, but I think it does make more sense for it to be a separate PR.

**Comment by donaldguy** [comment]
As an aside, if you want a more greenfield project for this sort of thing, I had contemplated trying to write a consul and maybe a VO connector for [guardian/alerta](https://github.com/guardian/alerta) but of course thats in your old pal Python so probably not interesting anymore amirite

**Comment by mboorstin** [comment]
I ended up submitting the upstream pull request at https://github.com/AcalephStorage/consul-alerts/pull/129, feel free to comment further on it.

**Comment by mboorstin** [comment]
Merged upstream; closing this.

**Pull request closed without merging**